UnityProject/Assets/Scripts/Transform/CustomNetTransform.cs

```diff
@@ -542,6 +545,22 @@ private void AdminRespawn()
 	{
 		PlayerManager.PlayerScript.playerNetworkActions.CmdAdminRespawn(gameObject);
 	}
+
+	// Checks if the object is occupiable and update occupant position if it's occupied (ex: a chair)
+	private void UpdateOccupant()
+	{
+		// Since we are at the transform level, we go up one level to get the parent
+		OccupiableDirectionalSprite occupiableDirectionalSprite = GetComponentInParent<OccupiableDirectionalSprite>();
+		if ((occupiableDirectionalSprite != null) && (occupiableDirectionalSprite.Occupant > 0))
+		{
+			NetworkIdentity networkIdentity = NetworkIdentity.spawned[occupiableDirectionalSprite.Occupant];
+			if (networkIdentity != null && networkIdentity.TryGetComponent<PlayerMove>(out var playerMove))
```

Comment:
You can avoid a GetComponent call here by caching PlayerScript of the buckled player in OccupiableDirectionalSprite whenever the buckled player net ID changes (just do it in OccupiableDirectionalSprite.syncOccupant, check if the game object with that net ID has a PlayerScript component and save it to a private field like buckledPlayerScript with a read-only public accessor like BuckledPlayerScript). Then you can access that via occupiableDirectionalSprite.BuckledPlayerScript and don't need TryGetComponent.

And PlayerScript already has the ability to cache and retrieve all of the other components on player via public accessors so that's pretty much the only component you need to have in order to avoid GetComponent calls.